internal/blockchain/ethereum/ethereum.go

```diff
@@ -418,29 +428,43 @@ func (e *Ethereum) SubmitBatchPin(ctx context.Context, operationID *fftypes.UUID
 		PayloadRef: batch.BatchPayloadRef,
 		Contexts:   ethHashes,
 	}
-	res, err := e.invokeContractMethod(ctx, e.instancePath, "pinBatch", signingKey, operationID.String(), input, tx)
+	res, err := e.invokeContractMethod(ctx, e.instancePath, "pinBatch", signingKey, operationID.String(), input)
 	if err != nil || !res.IsSuccess() {
 		return restclient.WrapRestErr(ctx, res, err, i18n.MsgEthconnectRESTErr)
 	}
 	return nil
 }
 
-func (e *Ethereum) InvokeContract(ctx context.Context, operationID *fftypes.UUID, signingKey string, location fftypes.Byteable, method *fftypes.FFIMethod, params map[string]interface{}) (interface{}, error) {
+func (e *Ethereum) InvokeContract(ctx context.Context, operationID *fftypes.UUID, signingKey string, location fftypes.Byteable, method *fftypes.FFIMethod, input map[string]interface{}) (interface{}, error) {
 	contractAddress, err := parseContractLocation(ctx, location)
 	if err != nil {
 		return nil, err
 	}
-	tx := &asyncTXSubmission{}
-	res, err := e.invokeContractMethod(ctx, fmt.Sprintf("contracts/%v", contractAddress.Address), method.Name, signingKey, operationID.String(), params, tx)
+	res, err := e.invokeContractMethod(ctx, fmt.Sprintf("contracts/%v", contractAddress.Address), method.Name, signingKey, operationID.String(), input)
 	if err != nil || !res.IsSuccess() {
 		return nil, restclient.WrapRestErr(ctx, res, err, i18n.MsgEthconnectRESTErr)
 	}
-	var result interface{}
-	err = json.Unmarshal(res.Body(), &result)
+	tx := &asyncTXSubmission{}
+	if err = json.Unmarshal(res.Body(), tx); err != nil {
+		return nil, err
+	}
+	return tx, nil
+}
```

Comment:
Does this change the existing behavior of what is returned when invoking a contract?

Reply:
It should not. For some reason we were unmarshaling twice - once via Resty's "SetResult" functionality, and once via manually parsing the Body. I kept the latter since it seems to give slightly more descriptive error messages.